20250314-RS-event-streaming.md

```diff
@@ -0,0 +1,695 @@
+  - `security`: Security configuration
+    - For remote collectors: `mtls: true` and `identity` with SPIFFE ID
+    - For local apps: `mtls: false`
+  - `filters`: Optional event filtering configuration
+- When configured, events are sent to:
+  - `OnEventAlpha1` endpoint for gRPC
+  - HTTP is not supported
+
+###### Collector Configuration
+- Collectors can be Dapr-enabled applications, but they don't have to be
+- To prevent recursive event collection:
+  - Runtime validation prevents a Dapr sidecar from being configured to send events to its own address
+
+#### Event Filtering
+- Can be configured per API using fully qualified API names:
+  - `invoke.dapr.io/v1`
+  - `state.dapr.io/v1`
+  - `publish.dapr.io/v1`
+  - `bindings.dapr.io/v1`
+  - `actors.dapr.io/v1`
+  - `secrets.dapr.io/v1`
+  - `configuration.dapr.io/v1`
+  - `metadata.dapr.io/v1`
+  - `workflows.dapr.io/v1`
+  - `lock.dapr.io/v1alpha1`
+  - `unlock.dapr.io/v1alpha1`
+  - `crypto.dapr.io/v1alpha1`
+  - `subtlecrypto.dapr.io/v1alpha1`
+  - `jobs.dapr.io/v1alpha1`
+  - `conversation.dapr.io/v1alpha1`
+  - `shutdown.dapr.io/v1` 
+- Can use `exclude` to exclude specific APIs when `*` is used for the `include`
+- When no filtering is specified, defaults to include `*` (all APIs)
+- Filtering can be applied to all sinks
+- Filtering can be configured either:
+  - Globally at the spec level (applies to all sinks)
+  - Per-sink (overrides global filters if present)
+
+#### Delivery Characteristics
+- Best-effort delivery (fire and forget)
+- Designed for high-volume event scenarios
+- No guaranteed delivery or acknowledgment
+- Ordering is not guaranteed, but timestamps can be used to infer timelines
+- After initial MVP, we can implement goroutines per filtered API type to send events to better guarantee ordering within filters for scale
+
+#### Code Changes
+
+##### Configuration:
+```yaml
+apiVersion: dapr.io/v1alpha1
+kind: Configuration
+spec:
+  events:
+    enabled: true  # Enable/disable all sinks
+    # Global filters (optional)
+    filters:
+      include: ["jobs.dapr.io/v1"]
+    sinks:
+    - address: "audit-collector:8080" # Actual network address
+      delivery: "transactional"  # or "best-effort" (default)
+      security: # required
+        mtls: true # default
+        identity: "spiffe://cluster.local/ns/cassie-app/sa/collector" # required if `mtls = true`
+      filters: # Overrides global filters
+        include: ["workflow.dapr.io/v1", "actors.dapr.io/v1"]
+    - address: "localhost:9090"
+      security:
+        mtls: false # optionally disable mtls
+        # identity can be nil if `mtls = false`
+      filters: # Overrides global filters
+        include: ["*"]
+```
+
+##### Configuration Validation:
+```go
+// Prevent self-referential event configurations
+if sinkAddress == ownAddress {
+    return fmt.Errorf("sink address cannot be the same as the Dapr sidecar's own address")
+}
```

Comment:
This can be the case when two processes share the same host address. They will be using different ports